src/views/catalog/wizard/tabs/network/components/list/NetworkInterfaceActions.tsx

```diff
@@ -50,6 +51,11 @@ const NetworkInterfaceActions: FC<NetworkInterfaceActionsProps> = ({
 
   const onDelete = useCallback(() => {
     const updatedVM = produceVMNetworks(vm, (draftVM) => {
+      if (isEmpty(draftVM.spec.template.spec.domain.devices.interfaces)) {
+        draftVM.spec.template.spec.domain.devices.autoattachPodInterface = false;
+        return;
+      }
+
       draftVM.spec.template.spec.networks = draftVM.spec.template.spec.networks.filter(
```

Comment:
I know you didn't change this, but you can you use the `getNetworks` selector her to make it `getNetworks(draftVM).filter...`?